### DISCUSSION TITLE
"-distort Resize" causes image edge to become translucent

### DISCUSSION THREAD

**Discussion by Ineth2** [ImageMagick/ImageMagick#3055] [Help] [answered]
I'm starting with a fully opaque image - for example, one generated like this:

```bash
magick -size 200x200 xc:red original.png
```
![original](https://user-images.githubusercontent.com/26445828/103383338-a6729400-4af2-11eb-9e9f-bb118d28b88c.png)

Then I use `-distort Resize` with colorspace conversion to up- or downscale the image:

```bash
magick original.png \
    -colorspace RGB \
    -filter Lanczos -define "filter:blur=0.88549061701764" \
    -distort Resize '100x100!' \
    -colorspace sRGB \
    +repage \
    PNG:resized.png
```
![resized](https://user-images.githubusercontent.com/26445828/103383382-d6219c00-4af2-11eb-8724-a79d5c4d307c.png)

(This method of scaling images is something I copied from some tutorial long ago.)

This used to work fine in ImageMagick 6, but with ImageMagick 7.0, the output image (`resized.png`) is **no longer fully opaque**: The pixels along the image edge are slightly translucent:

```bash
> identify -format "%[opaque]" resized.png
False
> identify -format "%k colors" resized.png
6 colors
```

I thought it might be a problem with the virtual-pixel setting, but adding `-virtual-pixel edge` before the `-distort` has not effect.

Any ideas what could cause this, and how to fix it?

---
_OS:_ Linux
_ImageMagick:_
```
Version: ImageMagick 7.0.10-46 Q16 x86_64 2020-12-06 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5) 
Delegates (built-in): bzlib cairo djvu fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png raqm raw rsvg tiff webp wmf x xml zlib
```

**Comment by fmw42** [comment]
I can reproduce this issue on IM 7.0.10.53 Q16 Mac OSX. *The changes are within a couple of pixels at the sides of the image*.  The verbose info shows the resize image to have a non-fully-opaque alpha channel with the following:

```
   Alpha:
      min: 237  (0.929412)
      max: 255 (1)
      mean: 254.758 (0.999053)
      standard deviation: 1.04122 (0.00408321)
      kurtosis: 47.748
      skewness: -5.69246
      entropy: 0.188391
  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 127.44 (0.499763)
      standard deviation: 0.260305 (0.0010208)
      kurtosis: -1.99998
      skewness: 5.05813e-05
      entropy: 0.0470978
  Colors: 6
  Histogram:
    9216: (255,0,0,255) #FF0000FF red
    384: (255,0,0,254) #FF0000FE srgba(255,0,0,0.996078)
    384: (255,0,0,250) #FF0000FA srgba(255,0,0,0.980392)
    8: (255,0,0,251) #FF0000FB srgba(255,0,0,0.984314)
    4: (255,0,0,253) #FF0000FD srgba(255,0,0,0.992157)
    4: (255,0,0,237) #FF0000ED srgba(255,0,0,0.929412)
 
```

This shows even more changes from pure opaque alpha and even changes in the color.

**`magick original.png -distort resize 100x100! x.png`**

**Comment by snibgo** [comment]
On Windows 8.1, the problem does not occur with v7.0.8-64.

It does occur with 7.0.10-54. This simpler command also shows the problem:
```
%IMG701010%magick -size 200x200 xc:red ^
    -distort Resize "100x100^!" ^
    PNG:r.png
```
This looks like an IM bug. A simple "-resize" does not have the problem.

**Reply by GeeMack** [reply]
When I include "-filter box" or "-filter point" the problem doesn't occur. I don't have a ready example, but I've run into a couple situations where scaling with "-distort SRT" doesn't seem to cooperate with "-filter". It might also be that I don't really understand it well enough to comment.

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow. 

In the mean-time, add `-alpha off` after your distort option.
